packages/kotti-ui/source/kotti-field-date/types.ts

```diff
@@ -1,45 +1,118 @@
+import dayjs from 'dayjs'
+import { z } from 'zod'
+
 import { KottiField } from '../kotti-field/types'
 
-export namespace Shared {
-	export interface Props<VALUE, PLACEHOLDER_TYPE = string | null>
-		extends KottiField.Props<VALUE, PLACEHOLDER_TYPE> {
-		maximumDate: string | null
-		minimumDate: string | null
-		shortcuts: ShortcutEntry<VALUE>[]
-	}
+import { DATE_FORMAT_REGEX, DATE_TIME_FORMAT_REGEX } from './constants'
 
-	export type ShortcutEntry<VALUE> = {
-		keepOpen?: boolean
-		label: string
-		value: VALUE
-	}
+export namespace Shared {
+	export const dateShortcutSchema = z.object({
+		keepOpen: z.boolean().optional(),
+		label: z.string(),
+		value: z.string().regex(DATE_FORMAT_REGEX),
+	})
+	export const dateTimeShortcutSchema = Shared.dateShortcutSchema.extend({
+		value: z.string().regex(DATE_TIME_FORMAT_REGEX),
+	})
 
+	export const valueSchema = z.string().nullable()
 	/**
 	 * ISO8601
 	 */
-	export type Value = string | null
+	export type Value = z.output<typeof valueSchema>
+
+	export const propsSchema = KottiField.propsSchema
+		.merge(
+			KottiField.potentiallySupportedPropsSchema.pick({
+				hideClear: true,
+			}),
+		)
+		.extend({
+			maximumDate: z
+				.string()
+				.refine((value) => dayjs(value).format('YYYY-MM-DD') === value)
+				.nullable()
+				.default(null),
+			minimumDate: z
+				.string()
+				.refine((value) => dayjs(value).format('YYYY-MM-DD') === value)
+				.nullable()
+				.default(null),
+			shortcuts: z.array(dateShortcutSchema).default(() => []),
+			placeholder: z.string().nullable().default(null),
+			value: valueSchema.default(null),
+		})
+	export type Props = z.input<typeof propsSchema>
+	export type PropsInternal = z.output<typeof propsSchema>
+
+	export namespace Range {
+		export const dateShortcutSchema = Shared.dateShortcutSchema.extend({
+			value: z.tuple([
+				z.string().regex(DATE_FORMAT_REGEX),
+				z.string().regex(DATE_FORMAT_REGEX),
+			]),
+		})
+		export const dateTimeShortcutSchema = Shared.dateShortcutSchema.extend({
+			value: z.tuple([
+				z.string().regex(DATE_TIME_FORMAT_REGEX),
+				z.string().regex(DATE_TIME_FORMAT_REGEX),
+			]),
+		})
+
+		export const valueSchema = z.tuple([
+			z.string().nullable(),
+			z.string().nullable(),
+		])
+
+		/**
+		 * [ ISO8601, ISO8601 ]
+		 */
+		export type Value = z.output<typeof valueSchema>
+
+		export const propsSchema = Shared.propsSchema.extend({
+			placeholder: z
+				.tuple([z.string().nullable(), z.string().nullable()])
+				.default([null, null]),
```

Comment:
```suggestion
				.default(() => [null, null]),
```

Reply:
Unlike arrays, tuples don't accept it

Zod errors if you do that (try it out locally)